File: Test110.py
__author__ = 'zyt'


# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    isBalance = True
    # @param {TreeNode} root
    # @return {boolean}
    def isBalanced(self, root):
        self.depth(root)
        return self.isBalance

    def depth(self, root):
        if not root:
            return 0
        left = self.depth(root.left)
        right = self.depth(root.right)
        if abs(left - right) > 1:
            self.isBalance = False
            return 0
        return 1 + max(left, right)

File: LeetCode_Java/src/Test110.java
/**
 * Created by zyt on 15/8/19 17:01.
 */
public class Test110 {
    /**
     * Definition for a binary tree node.
     */
    private class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode(int x) {
            val = x;
        }
    }

    static class Solution {
        boolean isBalanced = true;

        public boolean isBalanced(TreeNode root) {
            this.depth(root);
            return isBalanced;
        }

        public int depth(TreeNode root) {
            if (root == null) return 0;
            int left = 1 + this.depth(root.left);
            int right = 1 + this.depth(root.right);
            if (Math.abs(left - right) > 1) {
                isBalanced = false;
                return 0;
            }
            return Math.max(left, right);
        }
    }
}
